src/actions/NewSensorActions.js

```diff
@@ -28,11 +32,28 @@ const updateLoggingDelay = loggingDelay => ({
   payload: { loggingDelay },
 });
 
+const updateSensor = async (dispatch, getState) => {
+  const sensor = selectNewSensor(getState());
+  const { macAddress, logInterval } = sensor;
+
+  return dispatch(VaccineActions.startSetLogInterval({ macAddress, logInterval })).then(
```

Comment:
yeah, sorry. was still tidying and haven't pushed the changes. there's still something odd with the `BleService` - saying that sensors aren't connected but then throwing an error if you connect, saying 'already connected'.

Reply:
We recently changed the `BleService.connectAndDiscoverServices` to what it is now in mobile from this: https://github.com/openmsupply/msupply-cold-chain/blob/43f882c09732480d4b211536b98b2561de803070/src/common/services/Bluetooth/BleService.js#L18-L27

Because: I couldn't remember why it checks if you're connected, and disconnects if you are then reconnects- it seemed just like a silly thing. But I think I remember this is why. Could try giving this a go and seeing if it is more consistent.

Guess that's what comments are for, huh?

Reply:
thanks, sorted now and tested with various error conditions